presto-main/src/main/java/io/prestosql/sql/ExpressionUtils.java

```diff
@@ -225,7 +232,33 @@ public static Expression combineDisjunctsWithDefault(Metadata metadata, Collecti
             return TRUE_LITERAL;
         }
 
-        return disjuncts.isEmpty() ? emptyDefault : or(disjuncts);
+        ImmutableMultimap.Builder<Expression, Expression> builder = ImmutableMultimap.builder();
+        ImmutableList.Builder<Expression> expressionBuilder = ImmutableList.builder();
+        for (Expression disjunct : disjuncts) {
+            if (disjunct instanceof ComparisonExpression && ((ComparisonExpression) disjunct).getOperator() == EQUAL) {
+                ComparisonExpression comparisonExpression = (ComparisonExpression) disjunct;
+                builder.put(comparisonExpression.getLeft(), comparisonExpression.getRight());
```

Comment:
What if the expression `1 = a`? It will add `1` as the key and `a` as the value.

Reply:
But won't it be normalized here - `ExpressionInterpreter#visitComparisonExpression` ?